food-truck-frontend/src/pages/truck/edit/[truckId].tsx

```diff
@@ -1,140 +1,166 @@
+            console.log('Got error response code');
+          } else if (err.request) {
+            console.log('Did not receive Truck response');
+          } else {
+            console.log(err);
+          }
+
+          // Cancel
+          Router.replace("/");
+        });
     }
+  });
 
-    componentDidMount() {
-        api.get(`/truck/${this.props.truckId}`, {})
-            .then(res => {
-                // If the response indicates they own the truck, show them. Otherwise, kick out.
-                if (res && userCanEditTruck(res.data.userId)) {
-                    this.setState(res.data);
-                } else
-                    Router.replace("/");
-            }).catch(err => {
-        // If an error, log to console and kick out
-        if (err.response) {
-          console.log('Got error response code');
-        } else if (err.request) {
-          console.log('Did not receive Truck response');
-        } else {
-          console.log(err);
-        }
-
-        // Cancel
-        Router.replace("/");
-      });
-  }
-
-  onSubmit = (formData: any, response: AxiosResponse) => {
-    Router.replace(`/truck/${this.state.id}`);
+  const onSubmit = (formData: any, response: AxiosResponse) => {
+    Router.replace(`/truck/${state.id}`);
   }
 
-  onFail = (formData: any, response: AxiosResponse) => {
-    this.setState({
-      ...this.state,
+  const onFail = (formData: any, response: AxiosError) => {
+    setState({
+      ...state,
       message: `Failed to update truck details: ${JSON.stringify(response)}`
     });
   }
 
-  deleteTruck = () => {
-    api.delete(`/truck/delete/${this.state.id}`, {})
+  const deleteTruck = () => {
+    api.delete(`/truck/delete/${state.id}`, {})
       .then(res => Router.replace('/'))
       .catch(err => {
-        this.setState({
-          ...this.state,
+        setState({
+          ...state,
           message: `Failed to update truck details: ${JSON.stringify(err)}`
         })
       });
   }
+
+  if (state.userId == -1)
+    return <CircularProgress/>
+
+  return (
+    <Grid container alignItems="stretch">
+      <Grid container direction="row" alignItems="flex-start" className={classes.root}>
+        <Grid item className={classes.root}>
+          <Card>
+            <CardHeader title="Edit Truck Details"/>
+            <CardContent>
+              <Form submitMethod="PUT" submitUrl={'/truck/update'} onSuccessfulSubmit={onSubmit}
+                    onFailedSubmit={onFail}>
+                <TextField className={"hidden"} disabled label="Truck ID" name="truckId" defaultValue={state.id}/>
+                <TextField label="Truck Name" name="name" defaultValue={state.name}/>
+                <TextField label="Description" name="description"
+                           defaultValue={state.description}/>
+                <TextField label="Price Rating" name="priceRating"
+                           defaultValue={state.priceRating}/>
+                <MultiField
```

Comment:
Did you verify this was working? I know the MultiField can be finnicky sometimes.

Reply:
The `MultiField` for the tags seemed to work, but for some reason none of the fields are getting pre-populated with their default values now, even though the truck data comes back just fine. I'm not sure what changed from before, or if it never did this properly.

Reply:
I've found the issue. I apparently designed the component assuming that the default values never change after the component's creation. However, we're needing to reseed it here. Thus, we'd need to add the method `getDerivedStateFromProps` that overwrites the `MultiState.defaultState` with the new string array.